core/src/main/scala/cats/InvariantSemigroupal.scala

```diff
@@ -0,0 +1,28 @@
+package cats
+
+import simulacrum.typeclass
+
+/**
+  * [[InvariantSemigroupal]] is nothing more than something both invariant
+  * and Semigroupal. It comes up enough to be useful, and composes well
+  */
+@typeclass trait InvariantSemigroupal[F[_]] extends Semigroupal[F] with Invariant[F] { self =>
+
+   def composeApply[G[_]: Apply]: InvariantSemigroupal[λ[α => F[G[α]]]] =
+     new ComposedInvariantApplySemigroupal[F, G] {
+       def F = self
+       def G = Apply[G]
+     }
+
+
+}
+
+object InvariantSemigroupal extends SemigroupalArityFunctions {
+  def semigroup[F[_], A](implicit f: InvariantSemigroupal[F], sg: Semigroup[A]): Semigroup[F[A]] =
```

Comment:
The diff coverage is a bit low, one thing we could do is to add doctests for this `semigroup` method and the one in `ContravariantSemitgroupal`, which could also serve as examples. Not critically important though.